ruby/src/IceRuby/Operation.cpp

```diff
@@ -574,9 +574,7 @@ IceRuby::OperationI::unmarshalException(const vector<byte>& bytes, const Ice::Co
             volatile VALUE cls = CLASS_OF(ex);
             volatile VALUE path = callRuby(rb_class_path, cls);
             assert(TYPE(path) == T_STRING);
-            Ice::UnknownUserException e(__FILE__, __LINE__);
-            e.unknown = RSTRING_PTR(path);
-            throw e;
+            throw Ice::UnknownUserException{__FILE__, __LINE__, RSTRING_PTR(path)};
```

Comment:
Is this usage of RSTRING_PTR correct? RSTRING_PTR returns a pointer to the Ruby-allocated memory for the string, and the previous usage is indeed suspicious because the documentation states that the returned string is not guaranteed to be null-terminated.

The correct usage would be:
```
string(RSTRING_PTR(path), RSTRING_LEN(path))
```

Reply:
Created an issue.